src/main/kotlin/server/Connection.kt

```diff
@@ -34,8 +35,17 @@ class Connection(private val sendText: (String) -> Unit) {
             } else buffer.add(text)
         } ?: run {
             if (text.startsWith(";")) {
-                val code = text.substringAfter(";")
-                sendText(Compiler.eval(Context(this), "return $code"))
+                val eval = text.substringAfter(";")
+                val source = if (eval.startsWith(";")) "notifyConn(${eval.substringAfter(";")})" else eval
```

Comment:
I couldn't find an elegant way to get return values back for evals after I got strict about the stack.  So for now I have this hack -- single ; executes the code, double ;; prints the return value.